driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -57,17 +70,16 @@
         String srvHostDomain = srvHost.substring(srvHost.indexOf('.') + 1);
         List<String> srvHostDomainParts = asList(srvHostDomain.split("\\."));
         List<String> hosts = new ArrayList<String>();
-        InitialDirContext dirContext = createDnsDirContext();
         try {
-            String resourceRecordName = "_" + srvServiceName + "._tcp." + srvHost;
-            Attributes attributes = dirContext.getAttributes(resourceRecordName, new String[]{"SRV"});
-            Attribute attribute = attributes.get("SRV");
-            if (attribute == null) {
-                throw new MongoConfigurationException(format("No SRV records available for %s", resourceRecordName));
+            List<String> srvAttributeValues = dnsClient.getAttributeValues("_" + srvServiceName + "._tcp." + srvHost, "SRV");
+            if (srvAttributeValues == null) {
+                throw new MongoConfigurationException("No SRV records available for host " + "_mongodb._tcp." + srvHost);
             }
-            NamingEnumeration<?> srvRecordEnumeration = attribute.getAll();
-            while (srvRecordEnumeration.hasMore()) {
-                String srvRecord = (String) srvRecordEnumeration.next();
+            if (srvAttributeValues.isEmpty()) {
+                throw new MongoConfigurationException("Unable to find any SRV records for host " + srvHost);
+            }
```

Comment:
What is the reason for having slightly differently formulated error messages for `srvAttributeValues == null` and `srvAttributeValues.isEmpty()`? Note that the meaning of both messages is identical.

Reply:
Done